tools/integration-tests/tester/tests/consensus/consensus_conflict_spam_test.go

```diff
@@ -0,0 +1,185 @@
+	outputs = splitToAddresses(t, peer1, outputs[0], fundingKeyPair, addresses...)
+	const conflictRepetitions = 4
+	pairwiseOutputs := outputs[:(conflictRepetitions-1)*3+3]
+	tripletOutputs := outputs[len(pairwiseOutputs):]
+
+	txs := []*ledgerstate.Transaction{}
+	for i := 0; i < conflictRepetitions; i++ {
+		sendPairWiseConflicts(t, n.Peers(), pairwiseOutputs[i*3:i*3+3], keyPairs, &txs, i)
+		sendTripleConflicts(t, n.Peers(), tripletOutputs[i*3:i*3+3], keyPairs, &txs, i)
+	}
+	t.Logf("number of txs to verify is %d", len(txs))
+	verifyConfirmationsOnPeers(t, n.Peers(), txs)
+}
+
+func verifyConfirmationsOnPeers(t *testing.T, peers []*framework.Node, txs []*ledgerstate.Transaction) {
+	const unknownGoF = 10
+	for _, tx := range txs {
+		// current value signifies that we don't know what is the previous gof
+		var prevGoF gof.GradeOfFinality = unknownGoF
+		for i, peer := range peers {
+			var metadata *jsonmodels.TransactionMetadata
+			var err error
+			require.Eventually(t, func() bool {
+				metadata, err = peer.GetTransactionMetadata(tx.ID().Base58())
+				return err == nil && metadata != nil
+			}, 10*time.Second, time.Second, "Peer %s can't fetch metadata of tx %s. metadata is %v. Error is %w",
+				peer.Name(), tx.ID().Base58(), metadata, err)
+			t.Logf("GoF is %d for tx %s in peer %s", metadata.GradeOfFinality, tx.ID().Base58(), peer.Name())
+			if prevGoF != unknownGoF {
+				require.EqualValues(t, prevGoF, metadata.GradeOfFinality,
+					"Different gofs on tx %s between peers %s and %s", tx.ID().Base58(),
+					peers[i-1].Name(), peer.Name())
+			}
+			prevGoF = metadata.GradeOfFinality
+		}
+	}
+}
+
+/**
+sendPairWiseConflicts receives a list of outputs controlled by a peer with certain peer index.
+It send them all to addresses controlled by the next peer, but it does so several time to create pairwise conflicts.
+The conflicts are TX_B<->TX_A<->TX_C
+*/
+func sendPairWiseConflicts(t *testing.T, peers []*framework.Node, outputs ledgerstate.Outputs,
+	keyPairs map[string]*ed25519.KeyPair,
+	txs *[]*ledgerstate.Transaction, iteration int) {
+
+	t.Logf("send pairwise conflicts on iteration %d", iteration)
+	peerIndex := (iteration + 1) % len(peers)
+
+	// find target addresses
+	targetAddresses := determineTargets(peers, iteration)
+
+	tx1 := tests.CreateTransactionFromOutputs(t, peers[0].ID(), targetAddresses, keyPairs, outputs...)
+	tx2 := tests.CreateTransactionFromOutputs(t, peers[1].ID(), targetAddresses, keyPairs, outputs[0])
+	tx3 := tests.CreateTransactionFromOutputs(t, peers[2].ID(), targetAddresses, keyPairs, outputs[2])
+
+	*txs = append(*txs, tx1, tx2, tx3)
+
+	PostTransactions(t, peers, peerIndex, "pairwise conflicts", tx1, tx2, tx3)
+}
+
+/**
+Creates conflicts as so
+TX_A<->TX_B TX_B<->TX_C TX_C<->TX_A
+*/
+func sendTripleConflicts(t *testing.T, peers []*framework.Node, outputs ledgerstate.Outputs,
+	keyPairs map[string]*ed25519.KeyPair,
+	txs *[]*ledgerstate.Transaction, iteration int) {
```

Comment:
Do you need the concept of `iteration` here?

Reply:
yup

Reply:
Can you factor it out? It seems this function shouldn't need to know about how you are iterating from the caller.